### DISCUSSION TITLE
Some general newbie questions

### DISCUSSION THREAD

**Discussion by tobias992** [nginx/unit#1447] [Support] [answered]
Hi @all,

Currently we use a normal nginx + php-fpm stack for our wordpress websites. We have been testing nginx + unit for almost 2 weeks now. We use nginx as a proxy to deliver static files and firewall rules. We forward all php requests to unit. This works quite well so far. Performance improvements are definitely noticeable. However, there are still a few open questions:

1. Each PHP application seems to be isolated. It seems as if each app has its own dedicated secure OPcache.

2. Are the PHP applications so isolated that you don't need additional open_basedir settings or should you set it anyway?

3. Custom ‘.user.ini’ files directly in the application doc_root are not yet supported?

4. Due to the nginx proxy the php applications have a SERVER_ADDR variable in the name of the unit listener (unix socket). Is there a way to customise this?

Many thanks for your support...

**Comment by ac000** [comment, marked as answer]
Hi,

Glad you are liking Unit so far!

>  Each PHP application seems to be isolated. It seems as if each app has its own dedicated secure OPcache.

Each application runs in their own set of processes, the default being a single process but can be increased via the [processes](https://unit.nginx.org/configuration/#application-processes) setting.

If you look at the output of `ps -ef | grep unit` for example, you'll see for each application there is one prototype process and then 1 or more application processes which actually handle requests.

I'm not overly familiar with the OPcache, but I might have suspected that it would be per process/interpreter.

> Are the PHP applications so isolated that you don't need additional open_basedir settings or should you set it anyway?

By default each php application will have general access to the filesystem as governed by the user it is running as.

Unit does have various [isolation](https://unit.nginx.org/configuration/#process-isolation) capabilities (on Linux) via namespaces(7).

There is a [rootfs](https://unit.nginx.org/configuration/#conf-rootfs) option which lets you effectively set set the fileystem root for the application.

If restricting filesystem access is important to you then you should use something like open_basedir or rootfs.

> Custom ‘.user.ini’ files directly in the application doc_root are not yet supported?

Not that specifically, no. But you can specify per app PHP settings and also php.ini files to use. See [here](https://unit.nginx.org/configuration/#configuration-php-options).

> Due to the nginx proxy the php applications have a SERVER_ADDR variable in the name of the unit listener (unix socket). Is there a way to customise this?

Not currently.

**Reply by tobias992** [reply]
Thank you for the answers.
